readthedocs/doc_builder/python_environments.py

```diff
@@ -295,19 +295,27 @@ def venv_path(self):
         return os.path.join(self.project.doc_path, 'envs', self.version.slug)
 
     def setup_base(self):
-        site_packages = ''
+        """Create a virtualenv, invoking python -m virtualenv"""
+
+        # --no-download was removed because of the pip breakage,
+        # it was sometimes installing pip 20.0 which broke everything
+        # https://github.com/readthedocs/readthedocs.org/issues/6585
+
+        # Important not to add empty string arguments, see:
+        # https://github.com/readthedocs/readthedocs.org/issues/7322
```

Comment:
We can move these two notes into the docstring with `.. note::`

Reply:
Cool!